src/containers/PageContainer/index.tsx

```diff
@@ -295,129 +225,23 @@ const PageContainer: FC = ({ children }) => {
-							<FutureButton glow onClick={() => openWallet()}>
-								<div
-									style={{
-										display: 'flex',
-										justifyContent: 'center',
-										verticalAlign: 'center',
-										alignItems: 'center',
-									}}
-								>
-									<div
-										style={{
-											display: 'inline-block',
-											width: '10%',
-											margin: '0px',
-											padding: '0px',
-										}}
-									>
-										<Spinner />
-									</div>
-									<p
-										style={{
-											display: 'inline-block',
-											width: '100%',
-											verticalAlign: 'center',
-											height: '18px',
-											marginLeft: '15px',
-											whiteSpace: 'nowrap',
-										}}
-										className={styles.Message}
-									>
-										Trying to connect {localStorage.getItem('chosenWallet')}
-									</p>
-								</div>
-							</FutureButton>
-						)}
-						{!account && !localStorage.getItem('chosenWallet') && (
-							<>
-								<FutureButton glow onClick={openWallet}>
-									Connect {pageWidth > 900 && 'Wallet'}
-								</FutureButton>
-								<BuyTokenRedirect />
-							</>
-						)}
-						{account && !isSearchActive && (
-							<>
-								{/* Mint button */}
-								{isOwnedByUser && (
-									<FutureButton
-										style={{ padding: '0px 12px' }}
-										glow={account != null}
-										onClick={() => {
-											account != null
-												? openMint()
-												: addNotification('Please connect your wallet.');
-										}}
-										loading={loading}
-									>
-										MINT
-									</FutureButton>
-								)}
-								{/* Status / Long Running Operation Button */}
-								{showStatus ? (
-									<TooltipLegacy
-										content={<MintPreview onOpenProfile={openProfile} />}
-									>
-										<NumberButton
-											rotating={statusCount > 0}
-											number={statusCount}
-											onClick={() => {}}
-										/>
-									</TooltipLegacy>
-								) : null}
-
-								{/* Transfer Progress button */}
-								{transferring.length > 0 && (
-									<TooltipLegacy content={<TransferPreview />}>
-										<NumberButton
-											rotating={transferring.length > 0}
-											number={transferring.length}
-											onClick={() => {}}
-										/>
-									</TooltipLegacy>
-								)}
-
-								{/* Buy token from external urls */}
-								<BuyTokenRedirect walletConnected />
-
-								{/* Profile Button */}
-								<IconButton
-									onClick={openProfile}
-									style={{ height: 32, width: 32, borderRadius: '50%' }}
-									iconUri={userIcon}
-								/>
-
-								{/* TODO: Change the triple dot button to a component */}
-								<div className={styles.Dots} onClick={openWallet}>
-									<div></div>
-									<div></div>
-									<div></div>
-								</div>
-							</>
-						)}
-					</>
-				</TitleBar>
+
+				<Header
+					pageWidth={pageWidth}
+					znsDomain={znsDomain}
+					account={account}
+					openModal={openModal}
+				/>
```

Comment:
@stacy-hunt @colbr @domw30 @padala @akshatmittal This is just one element used for header component. We used so many codes here which is hard to read

Reply:
looks good, nice!